extensions/data-plane/integration-tests/src/test/java/org/eclipse/dataspaceconnector/dataplane/http/DataPlaneHttpPullIntegrationTests.java

```diff
@@ -0,0 +1,268 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.Header;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.HttpResponse;
+import org.mockserver.model.HttpStatusCode;
+import org.mockserver.model.MediaType;
+import org.mockserver.verify.VerificationTimes;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static io.restassured.RestAssured.given;
+import static org.eclipse.dataspaceconnector.common.testfixtures.TestUtils.getFreePort;
+import static org.mockserver.integration.ClientAndServer.startClientAndServer;
+import static org.mockserver.matchers.Times.once;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+import static org.mockserver.stop.Stop.stopQuietly;
+
+/**
+ * System Test for Data Plane HTTP extension.
+ */
+@ExtendWith(EdcExtension.class)
+public class DataPlaneHttpPullIntegrationTests {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Faker FAKER = new Faker();
+    private static final String AUTH_HEADER_KEY = HttpHeaderNames.AUTHORIZATION.toString();
+
+    // Validation API
+    private static final int VALIDATION_API_PORT = getFreePort();
+    private static final String VALIDATION_API_HOST = "http://localhost:" + VALIDATION_API_PORT;
+
+    // DPF API
+    private static final int DPF_CONTROL_API_PORT = getFreePort();
+    private static final int DPF_PUBLIC_API_PORT = getFreePort();
+    private static final String DPF_PUBLIC_API_HOST = "http://localhost:" + DPF_PUBLIC_API_PORT;
+    private static final String DPF_PUBLIC_PATH = "/public";
+    private static final String DPF_DATA_PATH = String.format("%s/data", DPF_PUBLIC_PATH);
+
+    // Data source API
+    private static final int HTTP_SOURCE_API_PORT = getFreePort();
+    private static final String HTTP_SOURCE_API_HOST = "http://localhost:" + HTTP_SOURCE_API_PORT;
+    private static final String SOURCE_AUTH_VALUE = FAKER.lorem().word();
+
+    /**
+     * HTTP Source mock server.
+     */
+    private static ClientAndServer httpSourceClientAndServer;
+
+    /**
+     * Validation API mock server.
+     */
+    private static ClientAndServer validationClientAndServer;
+
+    private static final Map<String, String> PROPS = Map.of(
+            "web.http.control.port", String.valueOf(DPF_CONTROL_API_PORT),
+            "web.http.public.port", String.valueOf(DPF_PUBLIC_API_PORT),
+            "web.http.public.path", DPF_PUBLIC_PATH,
+            "edc.controlplane.validation-endpoint", VALIDATION_API_HOST
+    );
+
+    @BeforeAll
+    public static void setUp() {
+        PROPS.forEach(System::setProperty);
```

Comment:
it's better to avoid system properties to inject configuration to the EdcExtension, it can be done with this call on a setup method:
```
@BeforeEach
void setup(EdcExtension extension) {
    extension.registerSystemExtension(ConfigurationExtension.class, (ConfigurationExtension) () -> ConfigFactory.fromMap(configuration));
}
```